packages/3dom/src/facade/three-js/three-dom-element-spec.ts

```diff
@@ -29,5 +29,27 @@ suite('facade/three-js/three-dom-element', () => {
       const element = new ThreeDOMElement(graft, object3D);
       expect(element.ownerModel).to.be.equal(graft.model);
     });
+
+    suite('names', () => {
+      test('ignores a Three.js-generated name', () => {
```

Comment:
I think this is the opposite of what we want. GLTFLoader populates .name from the gltf file, not .userData.name. We might want to add this to the end-to-end test above, to prove that names in the gltf file make it down to us.

Reply:
Actually, I think that it might be more complex that either of us are suggesting. In reviewing `RobotExpressive.glb`, there are names that are generated by Three.js that we should not be using (e.g., `Torso (Node) -> [ main (Primitive) , grey (Primitive) ]` is converted to `Torso (Group) -> [ Torso_1 (Mesh) -> main (Material), Torso_2 (Mesh) -> grey (Material) ]` in Three.js).

GLTFLoader stashes the original glTF name in `userData.name`. At least, it does this in some cases. That's why we are preferring it, because `Torso` is name we want to expose, and `Torso_1` is not (to refer back to the example). I'll take a closer look at `GLTFLoader` before this lands.

Reply:
Oh, interesting. I must have misread the GLTFLoader code. As long as we have a test for it that passes, I'm happy.